tests/regression/src/lib.rs

```diff
@@ -87,70 +93,202 @@ mod tests {
         .unwrap();
     }
 
-    /// Test which creates a TestPair from config using `rsa_4096_sha512`. Only measures a pair handshake. 
     #[test]
     fn test_rsa_handshake() {
         valgrind_test("test_rsa_handshake", |ctrl| {
             ctrl.stop_instrumentation();
-            // Example usage with RSA keypair (default)
             let keypair_rsa = CertKeyPair::default();
             let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
-            // Create a pair (client + server) using that config, start handshake measurement
             let mut pair = TestPair::from_config(&config);
-            // Assert a successful handshake
             ctrl.start_instrumentation();
             assert!(pair.handshake().is_ok());
             ctrl.stop_instrumentation();
             Ok(())
         })
         .unwrap();
     }
-    /// Function to run specified test using valgrind
+
     fn run_valgrind_test(test_name: &str) {
         let exe_path = std::env::args().next().unwrap();
+        let commit_hash = get_current_commit_hash();
+        let output_file = create_output_file_path(test_name, &commit_hash);
+        let command = build_valgrind_command(&exe_path, test_name, &output_file);
+
+        println!("Running command: {:?}", command);
+        execute_command(command);
+
+        let annotate_output = run_cg_annotate(&output_file);
+        save_annotate_output(&annotate_output, &commit_hash, test_name);
+
+        let count = find_instruction_count(&annotate_output)
+            .expect("Failed to get instruction count from file");
+
+        println!("Instruction count for {test_name}: {count}");
+    }
+
+    fn get_current_commit_hash() -> String {
+        let output = Command::new("git")
+            .args(["rev-parse", "HEAD"])
+            .output()
+            .expect("Failed to get commit hash");
+
+        if !output.status.success() {
+            panic!("Git command failed");
+        }
+
+        String::from_utf8(output.stdout)
+            .expect("Invalid UTF-8 in commit hash")
+            .trim()
+            .to_string()
+    }
+
+    fn create_output_file_path(test_name: &str, commit_hash: &str) -> String {
```

Comment:
I think the naming/comment could be more specific -> output file path for what? I think having very consistent naming across processing pipeline would be useful. Maybe something like
- `raw_profile`
- `annotated_profile`
- `diff_profile`

In general there is a lot of creating file paths and saving `std::process::Output` to that file. I think we could maybe abstract some of the functionality out so instead of having
- `run_cg_annotate_diff`
- `save_cg_annotate_diff`
You could just have `cg_annotated_diff` return the actual `std_out` string, and then you'd
```
std::fs::write_to_file(diff_profile_file_name(test_case), diff_profile)l
```
Or something like that.